### PR TITLE
Remove deprecated APIs up to 1.7.0

### PR DESCRIPTION
**Pull request by nk1506**

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/TestFileScanTaskParser.java [hunk 1]
```diff
@@ -48,28 +40,6 @@ public void testNullArguments() {
         .hasMessage("Invalid JSON string for scan task: null");
   }
 
-  @ParameterizedTest
-  @ValueSource(booleans = {true, false})
-  public void testFileScanTaskParser(boolean caseSensitive) {
-    PartitionSpec spec = TestBase.SPEC;
-    FileScanTask fileScanTask = createFileScanTask(spec, caseSensitive);
-    String jsonStr = FileScanTaskParser.toJson(fileScanTask);
```
Comment (findepi) [inline core/src/test/java/org/apache/iceberg/TestFileScanTaskParser.java:56]: Should this test be removed, or adapted to use a non-deprecated method?

**Review by findepi — COMMENTED** [review]
thanks for doing this cleanup!

File: core/src/test/java/org/apache/iceberg/TestFileScanTaskParser.java (hunk 1 above)
Comment (nk1506) [reply core/src/test/java/org/apache/iceberg/TestFileScanTaskParser.java:56]: There is already a [test](https://github.com/apache/iceberg/blob/main/core/src/test/java/org/apache/iceberg/TestFileScanTaskParser.java#L75) with new [ScanTaskParser](https://github.com/apache/iceberg/blob/72b39ab91dfa04d713552e70b009c24510a1cd07/core/src/main/java/org/apache/iceberg/ScanTaskParser.java)

File: core/src/test/java/org/apache/iceberg/TestFileScanTaskParser.java [hunk 2]
```diff
@@ -48,28 +40,6 @@ public void testNullArguments() {
...
-    assertThat(jsonStr).isEqualTo(fileScanTaskJsonWithoutTaskType());
-    FileScanTask deserializedTask = FileScanTaskParser.fromJson(jsonStr, caseSensitive);
-    assertFileScanTaskEquals(fileScanTask, deserializedTask, spec, caseSensitive);
-  }
-
-  /** Test backward compatibility where task-type field is absent from the JSON string */
-  @ParameterizedTest
-  @ValueSource(booleans = {true, false})
-  public void testFileScanTaskParserWithoutTaskTypeField(boolean caseSensitive) {
```
Comment (nk1506) [inline core/src/test/java/org/apache/iceberg/TestFileScanTaskParser.java:65]: Already a [test](https://github.com/apache/iceberg/blob/72b39ab91dfa04d713552e70b009c24510a1cd07/core/src/test/java/org/apache/iceberg/TestFileScanTaskParser.java#L65) with new [ScanTaskParser](https://github.com/apache/iceberg/blob/72b39ab91dfa04d713552e70b009c24510a1cd07/core/src/main/java/org/apache/iceberg/ScanTaskParser.java)

File: common/src/main/java/org/apache/iceberg/common/DynMethods.java [hunk 3]
```diff
@@ -163,10 +162,6 @@ private BoundMethod(UnboundMethod method, Object receiver) {
       this.receiver = receiver;
     }
 
-    public <R> R invokeChecked(Object... args) throws Exception {
```
Comment (nk1506) [inline common/src/main/java/org/apache/iceberg/common/DynMethods.java:166]: Thought to removed these unused methods after deprecating invokeChecked() . PLMK if it breaks any changes.

File: .palantir/revapi.yml [hunk 4]
```diff
@@ -1025,14 +1025,11 @@ acceptedBreaks:
...
     org.apache.iceberg:iceberg-core:
-    - code: "java.method.visibilityReduced"
-      old: "method void org.apache.iceberg.encryption.PlaintextEncryptionManager::<init>()"
-      new: "method void org.apache.iceberg.encryption.PlaintextEncryptionManager::<init>()"
-      justification: "Deprecations for 1.6.0 release"
     - code: "java.element.noLongerDeprecated"
       old: "method void org.apache.iceberg.encryption.PlaintextEncryptionManager::<init>()"
       new: "method void org.apache.iceberg.encryption.PlaintextEncryptionManager::<init>()"
-      justification: "Constructor became private as part of deprecations cleanup for 1.6.0 release"
+      justification: "Constructor became private as part of deprecations cleanup for\
```
Comment (nastra) [inline .palantir/revapi.yml:1031]: nit: unnecessary change

File: common/src/main/java/org/apache/iceberg/common/DynMethods.java (hunk 3 above)
Comment (nastra) [reply common/src/main/java/org/apache/iceberg/common/DynMethods.java:166]: this method wasn't marked as deprecated, so we can't remove it

File: common/src/main/java/org/apache/iceberg/common/DynConstructors.java [hunk 5]
```diff
@@ -82,14 +76,6 @@ public <R> R invoke(Object target, Object... args) {
       return (R) newInstance(args);
     }
 
-    @Override
-    @SuppressWarnings("unchecked")
```
Comment (nastra) [inline common/src/main/java/org/apache/iceberg/common/DynConstructors.java:86]: this method wasn't marked as deprecated?

File: .palantir/revapi.yml [hunk 6]
```diff
@@ -1056,6 +1053,92 @@ acceptedBreaks:
...
+      old: "method void org.apache.iceberg.rest.auth.OAuth2Util.AuthSession::<init>(java.util.Map<java.lang.String,\
+        \ java.lang.String>, java.lang.String, java.lang.String, java.lang.String,\
+        \ java.lang.String, java.lang.String)"
+      justification: "Removing deprecated method"
+    - code: "java.method.returnTypeChanged"
+      old: "method org.apache.iceberg.BaseMetastoreTableOperations.CommitStatus org.apache.iceberg.BaseMetastoreTableOperations::checkCommitStatus(java.lang.String,\
+        \ org.apache.iceberg.TableMetadata)"
+      new: "method org.apache.iceberg.BaseMetastoreOperations.CommitStatus org.apache.iceberg.BaseMetastoreTableOperations::checkCommitStatus(java.lang.String,\
+        \ org.apache.iceberg.TableMetadata)"
+      justification: "Removing deprecated method"
```
Comment (nastra) [inline .palantir/revapi.yml:1141]: can you update this message to `Removing deprecated code` instead of `method`?

File: common/src/main/java/org/apache/iceberg/common/DynConstructors.java (hunk 5 above)
Comment (nastra) [reply common/src/main/java/org/apache/iceberg/common/DynConstructors.java:86]: or did we forget to mark this method as deprecated?

File: common/src/main/java/org/apache/iceberg/common/DynMethods.java [hunk 7]
```diff
@@ -127,8 +125,9 @@ public String toString() {
     /** Singleton {@link UnboundMethod}, performs no operation and returns null. */
     private static final UnboundMethod NOOP =
         new UnboundMethod(null, "NOOP") {
+
```
Comment (nastra) [inline common/src/main/java/org/apache/iceberg/common/DynMethods.java:128]: nit: unrelated change?

File: common/src/main/java/org/apache/iceberg/common/DynMethods.java [hunk 8]
```diff
@@ -71,7 +69,7 @@ public <R> R invokeChecked(Object target, Object... args) throws Exception {
 
     public <R> R invoke(Object target, Object... args) {
       try {
-        return this.invokeChecked(target, args);
+        return invokeChecked(target, args);
```
Comment (nastra) [inline common/src/main/java/org/apache/iceberg/common/DynMethods.java:72]: nit: unrelated change?

**Review by nastra — COMMENTED** [review]
@nk1506 can you also please address the deprecations in `ContentCache`?

File: common/src/main/java/org/apache/iceberg/common/DynConstructors.java (hunk 5 above)
Comment (nk1506) [reply common/src/main/java/org/apache/iceberg/common/DynConstructors.java:86]: I don't see any usage of this method. Since abstract method has been removed. But we still have the [invokeChecked](https://github.com/apache/iceberg/pull/10818/files#diff-bef8d1afda79d9ab9186f9fe726389f6dc25b17b1edc95756ab91f4131216e33R55) method in use at super class. 
Do you think we should safely remove this or mark it as deprecated for next release ? 
WDYT?

Comment (nastra) [reply common/src/main/java/org/apache/iceberg/common/DynConstructors.java:86]: @findepi can you take a look at this one please?

File: common/src/main/java/org/apache/iceberg/common/DynMethods.java [hunk 9]
```diff
@@ -51,10 +51,8 @@ public static class UnboundMethod {
           (method == null || method.isVarArgs()) ? -1 : method.getParameterTypes().length;
     }
 
-    /** @deprecated since 1.6.0, will be removed in 1.7.0 */
-    @Deprecated // will become private
     @SuppressWarnings("unchecked")
-    public <R> R invokeChecked(Object target, Object... args) throws Exception {
+    private <R> R invokeChecked(Object target, Object... args) throws Exception {
```
Comment (findepi) [inline common/src/main/java/org/apache/iceberg/common/DynMethods.java:55]: > `// will become private`

my mistake
it was partially correct in a sense that it should stop being public API

but it seems i overlooked the fact the method is overridden (by `DynConstructors.Ctor`) in a way changing the behavior (invoking Method vs invoking a Constructor).
as such, it should become _package_-private, so that the override still works

File: common/src/main/java/org/apache/iceberg/common/DynConstructors.java (hunk 5 above)
Comment (findepi) [reply common/src/main/java/org/apache/iceberg/common/DynConstructors.java:86]: @nastra thanks for the ping. this was overlook. more about this -- https://github.com/apache/iceberg/pull/10818/files#r1706672859

**Review by nastra — APPROVED** [review]

**Comment by nk1506** [comment]
@findepi , Please let me know if there are more comments on this.

**Comment by findepi** [comment]
@nk1506 LGTM!
@nastra are you planning on taking the final look?

File: common/src/main/java/org/apache/iceberg/common/DynConstructors.java [hunk 10]
```diff
@@ -84,7 +78,7 @@ public <R> R invoke(Object target, Object... args) {
 
     @Override
     @SuppressWarnings("unchecked")
-    public <R> R invokeChecked(Object target, Object... args) throws Exception {
+    <R> R invokeChecked(Object target, Object... args) throws Exception {
```
Comment (nastra) [inline common/src/main/java/org/apache/iceberg/common/DynConstructors.java:81]: I don't think we can make this change here without going through a deprecation cycle. Please mark this as deprecated since 1.7.0 and we can remove/make it less visible in 1.8.0

**Review by nastra — APPROVED** [review]
